src/test/testAutopublish.ts

```diff
@@ -0,0 +1,209 @@
+
+const pubsub = new PubSub();
+
+describe('self-test', () => {
+  it('query works', () => {
+    const query = `
+      {
+        species(name: "Tiger"){
+          id
+        }
+      }
+    `;
+    return graphql(schema, query).then( res => {
+      return expect(res.data['species']['id']).to.equals(0);
+    });
+  });
+
+  it('create mutation works', () => {
+    const mutation = `
+      mutation {
+        createSpecies(name: "Eagle"){
+          id
+        }
+      }
+    `;
+    const query = `
+      {
+        species(name: "Eagle"){
+          id
+          name
+        }
+      }
+    `;
+    const expected = { id: 3, name: 'Eagle' };
+    return graphql(schema, mutation).then( data => {
+      return graphql(schema, query).then( res => {
+        return expect(res.data['species']).to.deep.equal(expected);
+      });
+    });
+  });
+
+  it('update mutation works', () => {
+    const mutation = `
+      mutation {
+        updateSpecies(id: 1, newName: "Meow"){
+          id
+          name
+        }
+      }
+    `;
+    const query = `
+      {
+        species(id: 1){
+          id
+          name
+        }
+      }
+    `;
+    const expected = { id: 1, name: 'Meow' };
+    return graphql(schema, mutation).then( data => {
+      return graphql(schema, query).then( res => {
+        return expect(res.data['species']).to.deep.equal(expected);
+      });
+    });
+  });
+
+  it('delete mutation works', () => {
+    const mutation = `
+      mutation {
+        deleteSpecies(id: 3){
+          id
+          name
+        }
+      }
+    `;
+    const query = `
+      {
+        species(id: 3){
+          id
+          name
+        }
+      }
+    `;
+    const expected: any = null;
+    return graphql(schema, mutation).then( data => {
+      return graphql(schema, query).then( res => {
+        return expect(res.data['species']).to.deep.equal(expected);
+      });
+    });
+  });
+});
+
+describe('Autopublish', () => {
+  it('publishes payloads to the channel on mutations', (done) => {
+    const added: any[] = [];
+    pubsub.subscribe('createSpecies', (data: any) => added.push(data));
+    const updated: any[] = [];
+    pubsub.subscribe('updateSpecies', (data: any) => updated.push(data));
+    const deleted: any[] = [];
+    pubsub.subscribe('deleteSpecies', (data: any) => deleted.push(data));
+
+    // autopublish
+    autopublishMutationResults(schema, pubsub);
+
+    // run one mutation of each, then check to make sure you got em all!
+    const mutation = `
+      mutation {
+        createSpecies(name: "Rhino"){
+          name
+        }
+        updateSpecies(id: 0, newName: "Penguin"){
+          name
+        }
+        deleteSpecies(id: 2){
+          name
+        }
+      }
+    `;
+    const expected = {
+      'added': { name: 'Rhino' },
+      'updated': { name: 'Penguin' },
+      'deleted': { name: 'Dog' },
+    };
+    graphql(schema, mutation).then( data => {
+      setTimeout(() => {
```

Comment:
You have a race condition here :\
since mutations runs on serial, i would add a callback to the last one that will do the expected statements and then call done().